plugins/multi-auth/main.go

```diff
@@ -134,8 +111,36 @@ func (r registerer) registerHandlers(ctx context.Context, extra map[string]inter
 
 }
 
-func init() {
-	fmt.Printf("Plugin: router handler \"%s\" loaded!!!\n", HandlerRegisterer)
+func main() {}
+
+// This logger is replaced by the RegisterLogger method to load the one from KrakenD
+var logger Logger = noopLogger{}
+
+func (registerer) RegisterLogger(v interface{}) {
+	l, ok := v.(Logger)
+	if !ok {
+		return
+	}
+	logger = l
+	logger.Info(fmt.Sprintf("[PLUGIN: %s] Logger loaded", HandlerRegisterer))
+}
+
+// Logger is an interface for logging functionality.
+type Logger interface {
+	Debug(v ...interface{})
+	Info(v ...interface{})
+	Warning(v ...interface{})
+	Error(v ...interface{})
+	Critical(v ...interface{})
+	Fatal(v ...interface{})
 }
 
-func main() {}
+// Empty logger implementation
+type noopLogger struct{}
+
+func (n noopLogger) Debug(_ ...interface{})    {}
```

Comment:
This is repeated 3 times, can we have the interface and an exported `NoopLogger` implementation at `plugins/logger.go`?

Reply:
It's tricky for maintaining KrakenD plugins here. Each folder under `plugins` is a complete plugin package with its own `go.mod`. If we do want to share a functionality across plugins, we can put it in the `x` package instead.

Reply:
Ah gotcha, I'd seen the modules but hadn't connected that each one would need to import that package. It was a minor non-blocking comment.

In fact, KrakenD points to [this](https://github.com/luraproject/lura/blob/master/logging/log.go#L17) interface, whose package includes a `NoOp` implementation